yt/sample_data/api.py

```diff
@@ -0,0 +1,172 @@
+    The output format is subject to change.
+
+    The output of this function is cached so it will only generate one request per session.
+    """
+
+    # it would be nicer to have an actual api on the yt website server,
+    # but this will do for now
+    api_url = "https://raw.githubusercontent.com/yt-project/website/master/data/datafiles.json"  # noqa
+
+    response = requests.get(api_url)
+
+    if not response.ok:
+        raise RuntimeError(
+            "Could not retrieve registry data. Please check your network setup."
+        )
+
+    website_json = response.json()
+    # this dict follows this schema: {frontend_name: {flat dataframe-like}}
+
+    columns = ["code", "filename", "size", "url", "description"]
+    website_table = pd.concat(pd.DataFrame(d) for d in website_json.values())[columns]
+
+    # add a int-type byte size column
+    website_table["byte_size"] = website_table["size"].apply(_parse_byte_size)
+
+    # normalize urls to match the local json
+    website_table["url"] = website_table["url"].apply(
+        lambda u: u.replace("http", "https")
+    )
+
+    # load local data
+    pooch_json = json.load(
+        pkg_resources.resource_stream("yt", "sample_data_registry.json")
+    )
+    pooch_table = pd.DataFrame(pooch_json.values())
+
+    # merge tables
+    unified_table = website_table.merge(pooch_table, on="url", how="outer")
+
+    # PR 3089
+    # ideally we should be able to do this, but it's not possible
+    # at the time of writing because fhe "filename" is incomplete
+    # see the companion comment in load_sample
+    # unified_table.set_index("filename", inplace=True)
+    # unified_table.index.rename("id", inplace=True)
+    return unified_table
+
+
+def _get_test_data_dir_path():
+    from yt.config import ytcfg
+
+    test_data_dir = Path(ytcfg.get("yt", "test_data_dir"))
+    return test_data_dir
+
+
+def lookup_on_disk_data(fn):
+    """
+    Look for data file/dir on disk.
+
+    Returns
+    -------
+
+    pathlib.Path to a file/dir matching fn if any
+
+    Raises
+    ------
+    FileNotFoundError
+    """
+    # this should be used in yt.loaders.load
+
+    path = Path(fn).expanduser()
+
+    if path.exists():
+        return path
+
+    err_msg = f"No such file or directory: '{fn}'."
+
+    data_dir = _get_test_data_dir_path()
+    if data_dir.is_dir():
+        alt_path = Path.joinpath(data_dir, fn)
+        if alt_path.exists():
+            return alt_path
+        err_msg += f"\n(Also tried '{alt_path}')."
+
+    raise FileNotFoundError(err_msg)
+
+
+# create a pooch.Pooch instance
+# pardon the poor variable naming: it is common practice to give those dog names...
+data_registry = get_data_registry_table()
+cache_storage = _get_test_data_dir_path() / "yt_download_cache"
+pooch_registry = {k: v["hash"] for k, v in sample_data_registry.items()}
+base_url = "https://yt-project.org/data/"
+poochie = pooch.create(path=cache_storage, base_url=base_url, registry=pooch_registry)
```

Comment:
```suggestion
poochie = pooch.pooch.create(path=cache_storage, base_url=base_url, registry=pooch_registry)
```

Reply:
I don't think that's correct
<img width="810" alt="Screenshot 2021-03-25 at 21 30 29" src="https://user-images.githubusercontent.com/14075922/112539467-546ba280-8db1-11eb-843d-8d66668d9286.png">

Reply:
It is for the on demand import version 

![image](https://user-images.githubusercontent.com/6745434/112540366-1609bf80-8d80-11eb-93da-eb069e93a813.png)

Reply:
Oh right, I forgot _some_ on_demand_import add an unnecessary layer of complexity. I strongly feel like modules obtained from `yt.utilities.on_demand_import` should behave as drop-in replacements, so I'll update _there_.

Reply:
(should be fixed now)